msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -286,7 +288,21 @@ public void setSliceParameters(final String sliceParameters) {
         apiEventBuilder.setAuthority(authorityURL.getProtocol() + "://" + authorityURL.getHost());
         Telemetry.getInstance().startEvent(telemetryRequestId, apiEventBuilder);
 
-        List<User> users = mTokenCache.getUsers(Authority.createAuthority(mAuthorityString, mValidateAuthority).getAuthorityHost(), mClientId, new RequestContext(UUID.randomUUID(), mComponent, telemetryRequestId));
+        final UUID correlationId = UUID.randomUUID();
+        initializeDiagnosticContext(correlationId.toString());
+
+        List<User> users = mTokenCache.getUsers(
+                Authority.createAuthority(
```

Comment:
can we wrap Authority.createAuthority(mAuthorityString, mValidateAuthority).getAuthorityHost() in a helper? I am sure we need it in a few places.